examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_inference_stage.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+import typing
+
+import srf
+from mlflow.tracking.client import MlflowClient
+
+from morpheus.config import Config
+from morpheus.messages.multi_ae_message import MultiAEMessage
+from morpheus.pipeline.single_port_stage import SinglePortStage
+from morpheus.pipeline.stream_pair import StreamPair
+
+from ..messages.multi_dfp_message import MultiDFPMessage
+from ..utils.model_cache import ModelCache
+from ..utils.model_cache import ModelManager
+from ..utils.model_cache import UserModelMap
+
+logger = logging.getLogger("morpheus.{}".format(__name__))
+
+
+class DFPInferenceStage(SinglePortStage):
+
+    def __init__(self, c: Config, model_name_formatter: str = ""):
+        super().__init__(c)
+
+        self._client = MlflowClient()
+        self._fallback_user = self._config.ae.fallback_username
+        self._model_name_formatter = model_name_formatter
+
+        self._users_to_models: typing.Dict[str, UserModelMap] = {}
+
+        self._model_cache: typing.Dict[str, ModelCache] = {}
+        self._model_cache_size_max = 10
+
+        self._cache_timeout_sec = 600
```

Comment:
We should probably expose these as constructor arguments.

Reply:
Agreed, but this is a low priority. Skipping for now